dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/tracker/imports/bundle/RelationshipImportTest.java

```diff
@@ -107,4 +108,20 @@ void successUpdateRelationships() throws IOException {
     assertThat(importReport.getStats().getCreated(), is(0));
     assertThat(importReport.getStats().getIgnored(), is(1));
   }
+
+  @Test
+  void shouldFailWhenTryingToUpdateADeletedRelationship() throws IOException {
+    TrackerImportParams trackerImportParams =
+        TrackerImportParams.builder().userId(userA.getUid()).build();
+    TrackerObjects trackerObjects = fromJson("tracker/relationships.json");
+    trackerImportService.importTracker(trackerImportParams, trackerObjects);
+
+    manager.delete(manager.get(Relationship.class, "Nva3Xj2j75W"));
+
+    trackerImportParams.setImportStrategy(TrackerImportStrategy.CREATE_AND_UPDATE);
+    ImportReport importReport =
+        trackerImportService.importTracker(trackerImportParams, trackerObjects);
+    assertHasError(importReport, ValidationCode.E4017);
+    assertThat(importReport.getStats().getIgnored(), is(2));
```

Comment:
why is the number 2?
we decided to stick to JUnit assertions instead of hamcrest https://github.com/dhis2/wow-backend/blob/master/guides/testing/testing.md#assertions